pkg/reconciler/buildrun/runner/taskrun.go

```diff
@@ -0,0 +1,491 @@
+			}
+		}
+	}
+}
+
+func updateBuildRunStatusWithOutputResult(buildRun *v1beta1.BuildRun, taskRunResult []pipelineapi.TaskRunResult) {
+	if buildRun.Status.Output == nil {
+		buildRun.Status.Output = &v1beta1.Output{}
+	}
+
+	for _, result := range taskRunResult {
+		switch result.Name {
+		case generateOutputResultName(imageDigestResult):
+			buildRun.Status.Output.Digest = result.Value.StringVal
+
+		case generateOutputResultName(imageSizeResult):
+			// TODO: Throw error that should not be retried?
+			if size, err := strconv.ParseInt(result.Value.StringVal, 10, 64); err == nil {
+				buildRun.Status.Output.Size = size
+			}
+		case generateOutputResultName(imageVulnerabilities):
+			buildRun.Status.Output.Vulnerabilities = getImageVulnerabilitiesResult(result)
+		}
+	}
+}
+
+func generateOutputResultName(resultName string) string {
+	return fmt.Sprintf("%s-%s", prefixParamsResultsVolumes, resultName)
+}
+
+func getImageVulnerabilitiesResult(result pipelineapi.TaskRunResult) []v1beta1.Vulnerability {
+	var vulns []v1beta1.Vulnerability
+	if len(result.Value.StringVal) == 0 {
+		return vulns
+	}
+
+	vulnerabilities := strings.Split(result.Value.StringVal, ",")
+	for _, vulnerability := range vulnerabilities {
+		vuln := strings.Split(vulnerability, ":")
+		severity := getSeverity(vuln[1])
+		vulns = append(vulns, v1beta1.Vulnerability{
+			ID:       vuln[0],
+			Severity: severity,
+		})
+	}
+	return vulns
+}
+
+func getSeverity(sev string) v1beta1.VulnerabilitySeverity {
+	switch strings.ToUpper(sev) {
+	case "L":
+		return v1beta1.Low
+	case "M":
+		return v1beta1.Medium
+	case "H":
+		return v1beta1.High
+	case "C":
+		return v1beta1.Critical
+	default:
+		return v1beta1.Unknown
+	}
+}
+
+func (t *TaskRunBuildRunner) updateBuildRunUsingTaskRunCondition(ctx context.Context, client client.Client, buildRun *v1beta1.BuildRun, trCondition *apis.Condition) error {
+	var reason, message string = trCondition.Reason, trCondition.Message
+	status := trCondition.Status
+
+	switch pipelineapi.TaskRunReason(reason) {
+	case pipelineapi.TaskRunReasonStarted:
+		fallthrough
+	case pipelineapi.TaskRunReasonRunning:
+		if buildRun.IsCanceled() {
+			status = corev1.ConditionUnknown // in practice the taskrun status is already unknown in this case, but we are making sure here
+			reason = v1beta1.BuildRunStateCancel
+			message = "The user requested the BuildRun to be canceled.  This BuildRun controller has requested the TaskRun be canceled.  That request has not been process by Tekton's TaskRun controller yet."
+		}
+	case pipelineapi.TaskRunReasonCancelled:
+		if buildRun.IsCanceled() {
+			status = corev1.ConditionFalse // in practice the taskrun status is already false in this case, bue we are making sure here
+			reason = v1beta1.BuildRunStateCancel
+			message = "The BuildRun and underlying TaskRun were canceled successfully."
+		}
+
+	case pipelineapi.TaskRunReasonTimedOut:
+		reason = "BuildRunTimeout"
```

Comment:
We have no constant for this ?

Reply:
Apparently not - my search for `BuildRunTimeout` only returns refrences to the string and not assignment to a constant. Promoting this to the API package is a good idea.